controller/Lists.php

```diff
@@ -292,29 +299,38 @@ public function saveLists(ValueCollectionService $valueCollectionService): void
             )
         );
 
-        foreach ($payload as $key => $value) {
-            if (preg_match('/^list-element_/', $key)) {
-                $encodedUri = preg_replace('/^list-element_[0-9]+_/', '', $key);
-                $uri = tao_helpers_Uri::decode($encodedUri);
-                $newUriValue = trim($payload["uri_$key"] ?? '');
-                $element = $elements->extractValueByUri($uri);
+        $listElements = array_filter($payload, function (string $key) {
+            return (bool)preg_match('/^list-element_/', $key);
+        },
+        ARRAY_FILTER_USE_KEY);
 
-                if ($element === null) {
-                    $elements->addValue(new Value(null, $newUriValue, $value));
+        foreach ($listElements as $key => $value) {
+            $encodedUri = preg_replace('/^list-element_[0-9]+_/', '', $key);
+            $uri = tao_helpers_Uri::decode($encodedUri);
+            $newUriValue = trim($payload["uri_$key"] ?? '');
+            $element = $elements->extractValueByUri($uri);
 
-                    continue;
-                }
+            if ($element === null) {
+                $elements->addValue(new Value(null, $newUriValue, $value));
 
-                $element->setLabel($value);
+                continue;
+            }
 
-                if ($newUriValue) {
-                    $element->setUri($newUriValue);
-                }
+            $element->setLabel($value);
+
+            if ($newUriValue) {
+                $element->setUri($newUriValue);
             }
         }
 
         try {
-            $this->returnJson(['saved' => $valueCollectionService->persist($elements)]);
+            $this->returnJson(
```

Comment:
We probably miss a proper error message for the user in case you try to add more than the limit :)

Reply:
You are right. Fixed in 7620ab28996ad009c8a1af75e90abef6a71c7159 :slightly_smiling_face: 

![image](https://user-images.githubusercontent.com/339792/147555975-07e1c3b7-ad34-4f39-92f2-a18ede2f3896.png)

Reply:
https://github.com/oat-sa/tao-core/pull/3303/files#r775844033